spec/unit/rom/support/inflector_spec.rb

```diff
@@ -34,22 +34,68 @@
   subject(:api) { ROM::Inflector }
 
   context 'with detected inflector' do
-    it_behaves_like 'an inflector'
+    before do
+      if api.instance_variables.include?(:@inflector)
+        api.remove_instance_variable(:@inflector)
+      end
+    end
+
+    it 'prefers ActiveSupport::Inflector' do
+      expect(api.inflector == ::ActiveSupport::Inflector).to be true
+    end
+  end
+
+  context 'with automatic detection' do
+    before do
+      if api.instance_variables.include?(:@inflector)
+        api.remove_instance_variable(:@inflector)
+      end
+    end
+
+    it 'automatically selects an inflector backend' do
+      expect(api.inflector).not_to be nil
+    end
   end
 
   context 'with ActiveSupport::Inflector' do
     before do
-      require 'active_support/inflector'
-      api.instance_variable_set(:@inflector, ::ActiveSupport::Inflector)
+      api.select_backend(:activesupport)
+    end
+
+    it 'is ActiveSupport::Inflector' do
+      expect(api.inflector == ::ActiveSupport::Inflector).to be true
     end
+
     it_behaves_like 'an inflector'
   end
 
   context 'with Inflecto' do
     before do
-      require 'inflecto'
-      api.instance_variable_set(:@inflector, ::Inflecto)
+      api.select_backend(:inflecto)
+    end
+
+    it 'is Inflecto' do
+      expect(api.inflector == ::Inflecto).to be true
     end
+
     it_behaves_like 'an inflector'
   end
+
+  context 'an unrecognized inflector library is selected' do
+    it 'raises a NameError' do
+      expect { api.select_backend(:foo) }.to raise_error(NameError)
+    end
+  end
+
+  context 'an inflector library cannot be found' do
+    before do
+      if api.instance_variables.include?(:@inflector)
+        api.remove_instance_variable(:@inflector)
+      end
+      stub_const("ROM::Inflector::BACKENDS", {})
+    end
+    it 'raises a LoadError' do
+      expect { api.inflector }.to raise_error(LoadError)
+    end
+  end
```

Comment:
Hmm what's the intention of this test? I mean, `BACKENDS` is a super-duper private implementation detail, we should assume nothing will touch it.